test/with_dummyserver/test_proxy_poolmanager.py

```diff
@@ -512,24 +501,146 @@ def test_proxy_pooling_ext(self):
             assert sc2 != sc3
             assert sc3 == sc4
 
-    @pytest.mark.timeout(0.5)
     @requires_network
-    def test_https_proxy_timeout(self):
-        with proxy_from_url("https://{host}".format(host=TARPIT_HOST)) as https:
+    @pytest.mark.parametrize(
+        ["proxy_scheme", "target_scheme", "use_forwarding_for_https"],
+        [
+            ("http", "http", False),
+            ("https", "http", False),
+            # 'use_forwarding_for_https' is only valid for HTTPS+HTTPS.
+            ("https", "https", True),
+        ],
+    )
+    def test_forwarding_proxy_request_timeout(
+        self, proxy_scheme, target_scheme, use_forwarding_for_https
+    ):
+        _should_skip_https_in_https(proxy_scheme, target_scheme)
```

Comment:
You're skipping testing ``use_forwarding_for_https`` in Python 2. I thought we still supported it but I might be misremembering?

Reply:
You're totally right, we do support that in Python 2. I've fixed that issue along with actually passing `use_forwarding_for_https` to the `ProxyManager` constructor :sweat_smile:

Local diff is:

```diff
diff --git a/test/with_dummyserver/test_proxy_poolmanager.py b/test/with_dummyserver/test_proxy_poolmanager.py
index 44d4558d..7b292a2d 100644
--- a/test/with_dummyserver/test_proxy_poolmanager.py
+++ b/test/with_dummyserver/test_proxy_poolmanager.py
@@ -514,7 +514,9 @@ class TestHTTPProxyManager(HTTPDummyProxyTestCase):
     def test_forwarding_proxy_request_timeout(
         self, proxy_scheme, target_scheme, use_forwarding_for_https
     ):
-        _should_skip_https_in_https(proxy_scheme, target_scheme)
+        _should_skip_https_in_https(
+            proxy_scheme, target_scheme, use_forwarding_for_https
+        )
 
         proxy_url = self.https_proxy_url if proxy_scheme == "https" else self.proxy_url
         target_url = "%s://%s" % (target_scheme, TARPIT_HOST)
@@ -567,13 +569,18 @@ class TestHTTPProxyManager(HTTPDummyProxyTestCase):
     def test_forwarding_proxy_connect_timeout(
         self, proxy_scheme, target_scheme, use_forwarding_for_https
     ):
-        _should_skip_https_in_https(proxy_scheme, target_scheme)
+        _should_skip_https_in_https(
+            proxy_scheme, target_scheme, use_forwarding_for_https
+        )
 
         proxy_url = "%s://%s" % (proxy_scheme, TARPIT_HOST)
         target_url = self.https_url if target_scheme == "https" else self.http_url
 
         with proxy_from_url(
-            proxy_url, ca_certs=DEFAULT_CA, timeout=SHORT_TIMEOUT
+            proxy_url,
+            ca_certs=DEFAULT_CA,
+            timeout=SHORT_TIMEOUT,
+            use_forwarding_for_https=use_forwarding_for_https,
         ) as proxy:
             with pytest.raises(MaxRetryError) as e:
                 proxy.request("GET", target_url)
@@ -608,11 +615,15 @@ class TestHTTPProxyManager(HTTPDummyProxyTestCase):
         ],
     )
     def test_https_proxy_tls_error(self, target_scheme, use_forwarding_for_https):
-        _should_skip_https_in_https("https", target_scheme)
+        _should_skip_https_in_https("https", target_scheme, use_forwarding_for_https)
 
         target_url = self.https_url if target_scheme == "https" else self.http_url
         proxy_ctx = ssl.create_default_context()
-        with proxy_from_url(self.https_proxy_url, proxy_ssl_context=proxy_ctx) as proxy:
+        with proxy_from_url(
+            self.https_proxy_url,
+            proxy_ssl_context=proxy_ctx,
+            use_forwarding_for_https=use_forwarding_for_https,
+        ) as proxy:
             with pytest.raises(MaxRetryError) as e:
                 proxy.request("GET", target_url)
             assert type(e.value.reason) == SSLError
@@ -635,7 +646,10 @@ class TestHTTPProxyManager(HTTPDummyProxyTestCase):
         ctx = ssl.create_default_context()
 
         with proxy_from_url(
-            proxy_url, proxy_ssl_context=proxy_ctx, ssl_context=ctx
+            proxy_url,
+            proxy_ssl_context=proxy_ctx,
+            ssl_context=ctx,
+            use_forwarding_for_https=use_forwarding_for_https,
         ) as proxy:
             with pytest.raises(MaxRetryError) as e:
                 proxy.request("GET", self.https_url)
@@ -746,10 +760,13 @@ class TestHTTPSProxyVerification:
         assert w[0].category == SubjectAltNameWarning
 
 
-def _should_skip_https_in_https(proxy_scheme, target_scheme):
+def _should_skip_https_in_https(
+    proxy_scheme, target_scheme, use_forwarding_for_https=False
+):
     if (
         sys.version_info[0] == 2
         and proxy_scheme == "https"
         and target_scheme == "https"
+        and use_forwarding_for_https is False
     ):
         pytest.skip("HTTPS-in-HTTPS isn't supported on Python 2")
```

Reply:
Thanks for including the local diff, made it easier to review!